File: Practica 3/Cola/Cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.n = n
        self.contador = 0

    def insert(self, e):
        if not self.isFull():
            self.fin = (self.fin + 1) % self.n
            self.arreglo[self.fin] = e
            self.contador += 1
        else:
            print("La cola está llena.")

    def remove(self):
        if not self.isEmpty():
            elemento = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            self.contador -= 1
            return elemento
        else:
            print("La cola está vacía.")
            return None

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        else:
            print("La cola está vacía.")
            return None

    def isEmpty(self):
        return self.contador == 0

    def isFull(self):
        return self.contador == self.n

    def size(self):
        return self.contador


cola = Cola(5)
cola.insert(10)
cola.insert(20)
print(cola.remove())  
print(cola.peek())    

File: Practica 3/Cola/Cola.java
public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int contador;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.contador = 0;
    }

    public void insert(long e) {
        if (!isFull()) {
            fin = (fin + 1) % n;
            arreglo[fin] = e;
            contador++;
        } else {
            System.out.println("La cola está llena.");
        }
    }

    public long remove() {
        if (!isEmpty()) {
            long elemento = arreglo[inicio];
            inicio = (inicio + 1) % n;
            contador--;
            return elemento;
        } else {
            System.out.println("La cola está vacía.");
            return -1;
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        } else {
            System.out.println("La cola está vacía.");
            return -1;
        }
    }

    public boolean isEmpty() {
        return contador == 0;
    }

    public boolean isFull() {
        return contador == n;
    }

    public int size() {
        return contador;
    }

    public static void main(String[] args) {
        Cola cola = new Cola(5);
        cola.insert(10);
        cola.insert(20);
        System.out.println(cola.remove());  
        System.out.println(cola.peek());    
    }
}
